nodejs/eks/cluster.ts

```diff
@@ -496,32 +510,48 @@ export function createCore(
     }
 
     // Create the EKS service role
-    let eksRole: pulumi.Output<aws.iam.Role>;
-    if (args.serviceRole) {
-        eksRole = pulumi.output(args.serviceRole);
-    } else {
-        eksRole = new ServiceRole(
+    let eksServiceRole: ServiceRole | undefined;
+    if (!args.serviceRole) {
+        const managedPolicies = ["AmazonEKSClusterPolicy"];
+
+        // EKS auto mode requires additional managed policies
+        // see https://docs.aws.amazon.com/eks/latest/userguide/auto-enable-existing.html#_cli
+        if (args.autoMode?.enabled) {
+            managedPolicies.push(
+                "AmazonEKSComputePolicy",
+                "AmazonEKSBlockStoragePolicy",
+                "AmazonEKSLoadBalancingPolicy",
+                "AmazonEKSNetworkingPolicy",
+            );
+        }
+
+        eksServiceRole = new ServiceRole(
             `${name}-eksRole`,
             {
-                service: "eks.amazonaws.com",
+                service: pulumi.interpolate`eks.${dnsSuffix}`,
                 description: "Allows EKS to manage clusters on your behalf.",
-                managedPolicyArns: [
-                    {
-                        id: "arn:aws:iam::aws:policy/AmazonEKSClusterPolicy",
-                        arn: pulumi.interpolate`arn:${partition}:iam::aws:policy/AmazonEKSClusterPolicy`,
-                    },
-                ],
+                managedPolicyArns: managedPolicies.map((policy) => ({
+                    id: `arn:aws:iam::aws:policy/${policy}`,
+                    arn: pulumi.interpolate`arn:${partition}:iam::aws:policy/${policy}`,
+                })),
                 tags: args.tags,
+                // EKS Auto Mode needs "sts:TagSession" in addition to the default "sts:AssumeRole"
+                assumeRoleActions: args.autoMode?.enabled
+                    ? ["sts:AssumeRole", "sts:TagSession"]
+                    : undefined,
```

Comment:
The comment makes it look like `undefined` should be `["sts:AssumeRole"]` instead?

Reply:
just add both regardless of auto mode - more policies are moving to use session tags

Reply:
`["sts:AssumeRole"]` is the default of `assumeRoleActions`

Reply:
Good to know @bryantbiggs, thanks!

Reply:
Changed it to add `sts:TagSession` independent of EKS Auto Mode now